### ISSUE TITLE
Current Ore Generation

### ISSUE THREAD

**Issue by belathus** [belathus/Wanderlust-Renewed#147]
I'm going to sum up the current ore gen here. 

While doing this, I noticed a few things: goethite and siderite overlap almost everywhere. Siderite, however, does not appear in wet biomes. Magnetite and Limonite have identical generation as well, as they both spawn in deserts and beaches. Magnetite has an additional vein that appears only in hot biomes. I'm thinking we should change Limonite to appear in different biomes than magnetite to prevent overlap, and siderite should probably spawn at a different Y level than goethite.

Format is ore: range, relative frequency, generation type. Additional notes are in the line below. Preferred biomes typically have triple the generation rate. Exceptions are noted.
Types of generation:
* Layered veins: decently common; motherlodes with branching veins. The default generation type that I use.
* Clouds: large but sparse groups of ore. Often, there will be a lot of ore in a cloud, but it won't appear to be a single vein. Often has extremely sparse hint veins on or near the surface (under grass) to indicate the presence of a cloud. Shield biomes often make it incredibly easy to find a cloud, as the surface is stone. By default, clouds are incredibly rare. They're often supplemented by another generation type.
* Vertical veins: the veins branch either straight up or straight down.
* Sparse veins: often large in diameter, but don't have a contiguous vein that you can easily follow. This makes the vein very hard to follow when mining, making the vein appear less rich than it actually is. When I use this generation type, the veins are often hundreds of blocks long and about 5-10 blocks in diameter, so they are almost always very rich.
* Pipe vein: a tube of ore that is filled with something else, usually something hazardous. Used to generate some of the most valuable ores, such as diamond and emerald.
* Clusters: similar to vanilla ore generation. Clusters of ore will be, typically, 6-12 blocks of ore in a tight group. I don't use this very often.

### Overworld
* **Aluminum**: 40 to 85; layered veins; 1.001 (common)
  * Mountains only: also spawns at 80 to 165
* **Apatite**: 56 to 184; sparse veins; 1.252 (common)
  * Preferred biome: forest
* **Aum**: 50 to 90, small clouds on grass; 0.005 (this is misleading; they're fairly common)
* **Blue Topaz**: 10 to 45, cloud distribution, 0.995 (common)
* **Cerublossom**: 50 to 90, small clouds on grass
* **Chimerite**: 10 to 45, cloud distribution, 0.995 (common)
* **Coal**: 0 to 128; sparse veins; 7.748 (extremely common)
* **Copper**: 36 to 78; layered veins; 1.416 (common)
  * Mountains: also spawns at 56 to 148
  * Also spawns in clouds and clusters
* **Dark Gem**: 6 to 66; Hint Veins; 20 (common, but only spawns a single ore at a time)
  * Preferred biome: spooky - triple frequency
  * Also spawns in clouds
* **Desert Nova**: 50 to 90, small clouds on sand; 0.005 (should be common in deserts)
* **Diamond**: 2-6, lava-filled pipe vein, 0.314 (rare and sparse)
* **Emerald**: 0-16, silverfish-filled pipe vein, 0.107 (very rare and sparse)
  * Preferred biome: mountains; 0.471 (uncommon. Still sparse)
* **Gold**: 0 to 32; layered veins; 0.708 (semi-common)
  * Mountains only: also spawns at 64 to 160
* **Iron**: Complicated. Spawns as magnetite, hematite, etc
  * Magnetite: spawns high in deserts and beaches as layered veins; 40 to 72; 2.0 (very common)
    * Spawns low in non-hot biomes as layered veins: 20 to 32; 2.0 (very common)
  * Hematite: spawns in cold biomes as layered veins; 40 to 72; 2.0 (very common)
  * Goethite: spawns everywhere as small deposits; 40 to 55; 0.769 (uncommon)
  * Limonite: spawns in deserts and beaches; 50 to 82; 2.0 (very common)
    * Can spawn in sandstone.
  * Siderite: spawns in hilly biomes as small deposits; 40 to 55; 0.769 (uncommon)
  * Taconite: spawns everywhere, but above sea-level; 70 to 90; 2.309 (very common)
    * In mountains, it spawns even higher: 100 to 200; same frequency
  * Bog Iron: only in swamps, as small deposits; 52 to 72; 1.252 (common)
  * Banded iron: spawns in plains biomes as clouds; 10 to 108; 1.707 (common)
* **Lapis**: 16+, vertical vein; 1.574 (common)
* **Lead**: 8 to 36; layered veins; 0.867 (common)
  * Mountains only: also spawns at 88 to 184
* **Liquid Essence**: 4 to 32, liquid-filled geode; 0.150 (fairly rare)
* **Menril Resin**: 4 to 32, liquid-filled geode; 0.150 (fairly rare)
* **Nickel**: 8 to 24; layered veins; 0.613 (uncommon)
  * Mountains only: also spawns at 130 to 244
  * 10% of the nickel spawned will be converted to platinum
* **Redstone**: 0 to 16+, vertical vein; 2.906 (very common)
* **Resonating Ore**: 0 to 16+; vertical veins; 1.416 (common)
* **Rutile**: 30 to 50, layered vein; 0.613 (uncommon)
  * Preferred biome: ocean; height changed to 20-40 in oceans
* **Silver**: spawns with lead, replacing lead ore 30% of the time
* **Tarma Root**: 6 to 62, small clouds on stone
* **Tin**: 36 to 60; layered veins; 1.416 (common)
  * Also spawns in clouds
* **Uranium**: 8 to 24; layered veins; 0.063 (very rare)
  * Preferred biome: wastelands; frequency: 0.613 (semi-common)
* **Vinteum**: 10 to 45, cloud distribution, 0.995 (common)
* **Wakebloom**: 50 to 90, small clouds on water
* **Zinc**: 36 to 60; layered veins; 1.416 (common)
  * Spawns in clouds, too

### The Nether
* **Nether Quartz**:10 to 108;  sparse veins; 6.247
* **Coal**: 4 to 124; layered vein (replaces air); 1.0
* **Iron**: 4 to 124; layered vein (replaces air); 0.333
* **Copper**: 4 to 124; layered vein (replaces air); 0.250
* **Lead/Silver**: 4 to 124; layered vein (replaces air); 0.350
* **Redstone**: 4 to 32; layered vein (replaces air); 0.150
* **Lapis**: 4 to 32; layered vein (replaces air); 0.210
* **Diamond**: 4 to 64; layered vein (replaces air); 0.150
  * Diamond is incredibly sparse (1%) in this vein
* **Emerald**: 5 to 35; layered vein (replaces air); 0.050
* **Prometheum**: 4 to 64; layered vein (replaces air); 0.150
* **Tiberium**: 4 to 64; layered vein (replaces air); 0.150
* **Tin**: 4 to 124; layered vein (replaces air); 0.150
* **Zinc**: 4 to 124; layered vein (replaces air); 0.150
* Heat Sand: -12 to 52; small deposit; 1.0
* Tainted Soil: -12 to 52; small deposit; 1.0
* **Redstone**: Under heat sand/tainted soil only; 0.5
* **Dimensional Shard**: Under heat sand/tainted soil only; 0.3
* **Uranium**: Under heat sand/tainted soil only; 0.5
* **Diamond/Emerald Geode**: 4 to 32; geode; 0.150
  * Stone base; diamond at 1.5%; emerald at 0.5%
* **Copper/Diamond Lava Geode**: 4 to 32; geode; 0.150
  * Abyssalstone base; copper at 6%; diamond at 1%
* **Iron Geode**: 24 to 72; geode; 0.150
  * Abyssalstone base; iron at 20%
* **Iron Lava Geode**: 24 to 72; geode; 0.075
  * Abyssalstone base; iron at 60%; filled with lava
* **Tin Geode**: 24 to 72; geode; 0.150
  * Abyssalstone base; tin at 8.5%; nitre at 12.5%
* **Tin Quicksand Geode**: 24 to 72; geode; 0.150
  * Abyssalstone base; tin at 15%; nitre at 25%
* **Copper/Gold Geode**: 24 to 72; geode; 0.150
  * Abyssalstone base; copper at 23%; gold at 4%
* **Gold/Copper Lava Geode**: 14 to 32; geode; 0.150
  * Abyssalstone base; copper at 6%; gold at 12%
* **Nickel/Platinum Antimatter Geode**: 91 to 119; geode; 0.150
  * Abyssalstone base; nickel at 7.5%; platinum at 1.5%

**Comment by Alan19** [comment]
Should this go in the wiki?

**Comment by belathus** [comment]
The purpose of this issue is to identify potential issues and resolve them. Once they're resolved, sure, we can put it in the wiki.

For example, one issue that I see here is that there is no platinum generation in the Overworld, but there is in the Nether. Similarly, there is no iridium generation anywhere, but I think we're planning on leaving it that way and either forcing players to process platinum to get iridium or go to an Advanced Rocketry asteroid or planet to get iridium ore.

**Comment by belathus** [comment]
Alright, made a bunch of changes. Most of the high-value nether ores are rarer. Mostly, the geodes have the same frequency, but the ores will be less dense inside the geode. Platinum now spawns in all nickel veins, which means that iridium should be obtainable via byproducts. We'll have to add iridium as a byproduct in non-Thermal Expansion ore processes such as the crusher and SAG mill.
